### PR TITLE
[GSOC][TMVA][SOFIE] Tanh ONNX operator added with the corresponding unit tests

### PR DESCRIPTION
**Pull request by Neel-Shah-29**
# This Pull request:  Adds the Tanh Activation function to SOFIE with the corresponding unit tests
1. Adds the Tanh ONNX operator, an activation function to SOFIE.
2. Adds the unit tests to validate the code.

## Checklist:

- [X] tested changes locally
- [ ] updated the docs (if necessary)

### CODE REVIEW

**Comment by bellenot** [comment]
@phsft-bot build

**Review by sanjibansg — CHANGES_REQUESTED** [review]
I think for the `tanh` function. we need to add the `cmath` library in the `Parse()` function.

cc: @lmoneta

**Comment by sanjibansg** [comment]
> I think for the `tanh` function. we need to add the `cmath` library.
> 
> cc: @lmoneta

Like this,
https://github.com/root-project/root/blob/7456f8da40944b7c1c627f19a35770eae7e16b92/tmva/sofie_parsers/src/RModelParser_ONNX.cxx#L1417

File: tmva/sofie/inc/TMVA/ROperator_Tanh.hxx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+   void Initialize(RModel& model){
+       //input must be a graph input, or already initialized intermediate tensor
+      if (model.CheckIfTensorAlreadyExist(fNX) == false){
+        throw std::runtime_error("TMVA SOFIE Tanh Op Input Tensor is not found in model");
+      }
+      fShape = model.GetTensorShape(fNX);
+      model.AddIntermediateTensor(fNY, model.GetTensorType(fNX), fShape);
+      //use the cmath library for the hyperbolic tan function
+      model.AddNeededStdLib("cmath");
```
Comment (sanjibansg) [inline tmva/sofie/inc/TMVA/ROperator_Tanh.hxx:46]: Ahh sorry, I see this is here now. Will it be better to move this to the `Parse()` function where we add the libraries and routines for this cases, just to keep a standard format.

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

File: tmva/sofie/inc/TMVA/ROperator_Tanh.hxx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+   void Initialize(RModel& model){
+       //input must be a graph input, or already initialized intermediate tensor
+      if (model.CheckIfTensorAlreadyExist(fNX) == false){
+        throw std::runtime_error("TMVA SOFIE Tanh Op Input Tensor is not found in model");
+      }
+      fShape = model.GetTensorShape(fNX);
+      model.AddIntermediateTensor(fNY, model.GetTensorType(fNX), fShape);
+      //use the cmath library for the hyperbolic tan function which is added in the Parse function
+      // model.AddNeededStdLib("cmath");
```
Comment (sanjibansg) [inline tmva/sofie/inc/TMVA/ROperator_Tanh.hxx:46]: you don't need to comment this out. Just remove this line of code if it isn't needed.

**Review by sanjibansg — CHANGES_REQUESTED** [review]

File: tmva/sofie/inc/TMVA/ROperator_Tanh.hxx (hunk 2 above)
Comment (Neel-Shah-29) [reply tmva/sofie/inc/TMVA/ROperator_Tanh.hxx]: okay done.

**Review by sanjibansg — APPROVED** [review]

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

**Review by lmoneta — APPROVED** [review]
LGTM!
Thank you for this contribution!

**Pull request merged**